src/macros.jl

```diff
@@ -592,6 +602,9 @@ function _constraint_macro(args, macro_name::Symbol, parsefun::Function)
     # Strategy: build up the code for add_constraint, and if needed
     # we will wrap in loops to assign to the ConstraintRefs
     refcall, idxvars, idxsets, condition = _build_ref_sets(c, variable)
+    if any(_expr_is_splat.(idxsets))
+        _error("cannot use splatting operator `...`.")
```

Comment:
Nit: I would clarify on the error message that you can't use the splatting operator *in the definition of an index set*. There are other cases where splatting works inside a macro.